File: Prácticas/P1/src/SuppliesPackage.java
/**
 * @author elena
 * @brief Representa un paquete de suministros para una estación espacial. Puede contener
 * armamento, combustible y/o energía para los escudos.
*/
class SuppliesPackage{
    private float ammoPower, fuelUnits, shieldPower;

    SuppliesPackage(float ammo, float fuel, float power){
        ammoPower= ammo;
        fuelUnits= fuel;
        shieldPower= power;
    }
    
    SuppliesPackage(SuppliesPackage s){
        this(s.getAmmoPower(), s.getFuelUnits(), s.getShieldPower());
    }
    
    public float getAmmoPower(){
        return ammoPower;
    }
    
    public float getFuelUnits(){
        return fuelUnits;
    }
    
    public float getShieldPower(){
        return shieldPower;
    }
    
}

File: Prácticas/P1/ruby/lib/supplies_package.rb
=begin
Representa a un paquete de suministros para una estación espacial. Puede contener
armamento, combustible y/o energía para los escudos
=end

module Deepspace
  class SuppliesPackage
    @ammoPower
    @fuelUnits
    @shieldPower
    
    def initialize(ammo, fuel, power)
      @ammoPower= ammo
      @fuelUnits= fuel
      @shieldPower= power
    end
    
    def newCopy(s)
      SuppliesPackage.new(s.ammoPower, s.fuelUnits, s.shieldPower)
    end
    
    attr_reader :ammoPower, :fuelUnits, :shieldPower
  end
end